### DISCUSSION TITLE
Slot issue

### DISCUSSION THREAD

**Discussion by gnorb** [advanced-chat/advanced-chat-components#494] [General]
Hi. According to docs, it's the way to use slots in Vue:
`<div slot="room-header">This is a new room header</div>`
  
  It's not working in Vue 3, obviously. So I changed it with:
  `<template #room-header>This is a new room header</template>`
  
  It's not working too. The error message I receive:
  `[plugin:vite:vue] Codegen node is missing for element/if/for node. Apply appropriate transforms first.`
  
  What is going on? It's Vue3 Composition API with Vite and NODE v.18.18.1.
  
  PS. 
  ```js   
  <vue-advanced-chat
      height="100%"
      :current-user-id="userID"
      :rooms="JSON.stringify(rooms)"
      :messages="JSON.stringify(messages)"
      :room-actions="JSON.stringify(roomActions)"
      rooms-loaded
      :messages-loaded="messagesLoaded"
      @send-message="sendMessage"
    >
      <template #room-header>This is a new room header</template>
    </vue-advanced-chat>```

**Comment by gnorb** [comment]
I fought with it about hour, seven minutes after posting it, I found solution :) The solution is, add:

```js
  rules: {
    "vue/no-deprecated-slot-attribute": "off",
  },
```

to your `.eslint.cjs`  file and use slots as wrote in docs.

**Reply by aioya** [reply]
big savior, thank you
